packages/mds-audit-service/service/validators.ts

```diff
@@ -14,62 +14,80 @@
  * limitations under the License.
  */
 
-import { schemaValidator } from '@mds-core/mds-schema-validators'
+import { SchemaValidator } from '@mds-core/mds-schema-validators'
 import { AUDIT_EVENT_TYPES, VEHICLE_EVENTS } from '@mds-core/mds-types'
-import Joi from 'joi'
 import { AuditAttachmentDomainModel, AuditDomainModel, AuditEventDomainModel } from '../@types'
 
+const uuidSchema = { type: 'string', format: 'uuid' }
+const enumSchema = <T>(enumType: T[]) => ({ type: 'string', enum: enumType })
+
 export const { validate: validateAuditDomainModel, isValid: isValidAuditDomainModel } =
-  schemaValidator<AuditDomainModel>(
-    Joi.object<AuditDomainModel>()
-      .keys({
-        audit_trip_id: Joi.string().uuid().required(),
-        audit_device_id: Joi.string().uuid().required(),
-        audit_subject_id: Joi.string().required(),
-        provider_id: Joi.string().uuid().required(),
-        provider_name: Joi.string().required(),
-        provider_vehicle_id: Joi.string().required(),
-        provider_device_id: Joi.string().uuid().allow(null)
-      })
-      .unknown(false)
-  )
+  SchemaValidator<AuditDomainModel>({
+    $id: 'Audit',
+    type: 'object',
+    properties: {
+      audit_trip_id: uuidSchema,
+      audit_device_id: uuidSchema,
+      audit_subject_id: { type: 'string' },
+      provider_id: uuidSchema,
+      provider_name: { type: 'string' },
+      provider_vehicle_id: { type: 'string' },
+      provider_device_id: uuidSchema
```

Comment:
Nullable in the original schema:
```suggestion
      provider_device_id: { ...uuidSchema, nullable: true, default: null }
```